systemd/systemd.go

```diff
@@ -246,8 +247,20 @@ type Systemd interface {
 	Unmask(service string) error
 }
 
-// A Log is a single entry in the systemd journal
-type Log map[string]string
+// A Log is a single entry in the systemd journal.
+// In almost all cases, the strings map to a single string value, but as per the
+// manpage for journalctl, under the json format,
+//
+//    Journal entries permit non-unique fields within the same log entry. JSON
+//    does not allow non-unique fields within objects. Due to this, if a
+//    non-unique field is encountered a JSON array is used as field value,
+//    listing all field values as elements.
+//
+// as such, we sometimes get array values which need to be handled differently,
+// and additionally if systemd thinks the value of a string is not safe ascii,
```

Comment:
Perhaps we could quote the bit from the manpage:

```
Fields containing non-printable or non-UTF8 bytes are
encoded as arrays containing the raw bytes individually
formatted as unsigned numbers.
```

Reply:
added